packages/p2p/src/components/my-profile/my-profile.jsx

```diff
@@ -1,70 +1,308 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import { Field, Form, Formik } from 'formik';
+import { Button, Icon, Input, Loading, Table, Popover } from '@deriv/components';
+import { requestWS } from 'Utils/websocket';
 import { localize } from 'Components/i18next';
-import { toSentenceCase } from 'Utils/string';
+import FooterActions from 'Components/footer-actions/footer-actions.jsx';
+import { textValidator } from 'Utils/validations';
+import { generateHexColourFromNickname, getShortNickname } from 'Utils/string';
 import './my-profile.scss';
 
-// TODO: This is just mockup data. it will be removed after Adding BE API.
-const user_data = {
-    first_name: 'johny',
-    last_name: 'bravo',
-    security_deposit: 0,
-    buy_trades: 23,
-    sell_trades: 25,
-    past_month_trades: 42,
-    past_month_completion_rate: 99,
-    average_release_time: 5,
-    average_release_unit: 'min',
-};
+const MyProfile = () => {
+    const [advertiser_info, setAdvertiserInfo] = React.useState({});
+    const [has_poa, setHasPoa] = React.useState(false);
+    const [has_poi, setHasPoi] = React.useState(false);
+    const [is_loading, setIsLoading] = React.useState(true);
+    const [is_mounted, setIsMounted] = React.useState(false);
+    const [nickname, setNickname] = React.useState(null);
+    const [stats, setStats] = React.useState({});
+    const {
+        contact_info,
+        daily_buy_limit,
+        daily_sell_limit,
+        default_advert_description,
+        id,
+        payment_info,
+    } = advertiser_info;
+    const { buy_orders_count, sell_orders_count, total_orders_count } = stats;
 
-const Cell = ({ title, value, upper_sub_value, lower_sub_value, dimension }) => {
-    return (
-        <div className='my-profile__cell'>
-            <div className='my-profile__cell-data'>
-                <div className='my-profile__cell-data-main'>
-                    {value}
-                    <sub className='my-profile__cell-data-dimension'>{dimension}</sub>
-                </div>
-                <div className='my-profile__cell-data-sub'>
-                    <div className='my-profile__cell-data-sub-upper'>{upper_sub_value}</div>
-                    <div className='my-profile__cell-data-sub-lower'>{lower_sub_value}</div>
-                </div>
-            </div>
-            <h2 className='my-profile__cell-title'>{title}</h2>
-        </div>
-    );
-};
+    React.useEffect(() => {
+        setIsMounted(true);
+        return () => setIsMounted(false);
+    }, []);
 
-Cell.propTypes = {
-    dimension: PropTypes.string,
-    lower_sub_value: PropTypes.string,
-    title: PropTypes.string,
-    upper_sub_value: PropTypes.string,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-};
+    React.useEffect(() => {
+        if (is_mounted) {
+            advertiserAccountStatus();
+            advertiserStats();
+            advertiserInfo();
```

Comment:
Could we prefix all of these with `get` for clarity, e.g. `getAdvertiserAccountStatus`